crates/noirc_evaluator/src/ssa/inline.rs

```diff
@@ -227,56 +265,100 @@ pub fn inline_in_block(
                     //we need to find the corresponding result instruction in the target block (using ins.rhs) and replace it by ins.lhs
                     for (i, value) in values.iter().enumerate() {
                         if ctx
-                            .get_result_instruction_mut(target_block_id, call_id, i as u32)
+                            .get_result_instruction_mut(stack_frame.block, call_id, i as u32)
                             .is_some()
                         {
-                            ctx.get_result_instruction_mut(target_block_id, call_id, i as u32)
+                            ctx.get_result_instruction_mut(stack_frame.block, call_id, i as u32)
                                 .unwrap()
                                 .mark = Mark::ReplaceWith(*value);
                         }
                     }
                     let call_ins = ctx.get_mut_instruction(call_id);
                     call_ins.mark = Mark::Deleted;
                 }
-                Operation::Call(..) => {
+                Operation::Call { .. } => {
                     *nested_call = true;
-                    let new_ins = new_cloned_instruction(clone, target_block_id);
+                    let new_ins = new_cloned_instruction(clone, stack_frame.block);
                     push_instruction(ctx, new_ins, stack_frame, inline_map);
                 }
                 Operation::Load { array_id, index } => {
                     //Compute the new address:
                     //TODO use relative addressing, but that requires a few changes, mainly in acir_gen.rs and integer.rs
                     let b = stack_frame.get_or_default(*array_id);
                     let offset = ctx.mem[b].adr as i32 - ctx.mem[*array_id].adr as i32;
-                    let index_type = ctx[*index].get_type();
-                    let offset_id =
-                        ctx.get_or_create_const(FieldElement::from(offset as i128), index_type);
-
-                    let add =
-                        node::Binary { operator: node::BinaryOp::Add, lhs: offset_id, rhs: *index };
-                    let adr_id = ctx.new_instruction(Operation::Binary(add), index_type)?;
+                    let adr_id = if *index == NodeId::dummy() {
+                        NodeId::dummy()
+                    } else {
+                        let index_type = ctx[*index].get_type();
+                        let offset_op = if offset < 0 {
```

Comment:
Why is this additional negative offset branch needed? New functionality or bugfix?

Reply:
It's a bugfix, basically the safeguard I added in the constant folding (in the `wrapping()` function) are not compatible with unsigned numbers. This should be fixed as well.
In any cases, this offset stuff is becoming annoying. I will remove it and use an absolute addressing once I am done with the IFs.